CMakeLists.txt

```diff
@@ -40,14 +40,20 @@ endif()
 project(falcosecurity-libs)
 
 option(USE_BUNDLED_DEPS "Enable bundled dependencies instead of using the system ones" ON)
-option(MINIMAL_BUILD "Produce a minimal build with only the essential features (no eBPF probe driver, no kubernetes, no mesos, no marathon and no container metadata)" OFF)
 option(MUSL_OPTIMIZED_BUILD "Enable if you want a musl optimized build" OFF)
 option(USE_BUNDLED_DRIVER "Use the driver/ subdirectory in the build process (only available in Linux)" ON)
 option(ENABLE_DRIVERS_TESTS "Enable driver tests (bpf, kernel module, modern bpf)" OFF)
 option(ENABLE_LIBSCAP_TESTS "Enable libscap unit tests" OFF)
 option(BUILD_SHARED_LIBS "Build libscap and libsinsp as shared libraries" OFF)
 option(ENABLE_VM_TESTS "Enable driver sanity tests" OFF)
 
+option(MINIMAL_BUILD "Produce a minimal build with only the essential features (no eBPF probe driver, no kubernetes, no mesos, no marathon and no container metadata)" OFF)
```

Comment:
just a minor note on the fly on which maybe we want to start discussing:
* What about having a unique `MINIMAL_BUILD`? Is not clear to me why bpf was excluded in principle :thinking: 
* What about enabling also the modern probe in `MINIMAL_BUILD`? In the end is just another ebpf driver and could be really useful for example in Falco with minimal build because we can test it even if we have no prebuilt drivers

WDYT? @falcosecurity/libs-maintainers

Reply:
> * What about enabling also the modern probe in `MINIMAL_BUILD`?

:+1: 

We can also think of including `zlib` in the minimal build. I don't see it as a huge issue. wdyt?
cc @gnosek

Reply:
> * What about having a unique `MINIMAL_BUILD`? Is not clear to me why bpf was excluded in principle 🤔

AFAICT, ebpf was excluded to allow for building without linking to zlib, doesn't seem to be any other reason for this. If we can directly add eBPF into the minimal build, this PR will become a lot smaller.


> * What about enabling also the modern probe in `MINIMAL_BUILD`?

Because of how everything is setup, it's already possible to set `MINIMAL_BUILD` and `BUILD_LIBSCAP_MODERN_BPF` to `ON` and everything will be setup and work correctly, so no change is needed for this to work (at least when I tried it, it worked this way).

Reply:
> Because of how everything is setup, it's already possible to set MINIMAL_BUILD and BUILD_LIBSCAP_MODERN_BPF to ON and everything will be setup and work correctly, so no change is needed for this to work (at least when I tried it, it worked this way).

oh, great news! never tried! thanks

> AFAICT, ebpf was excluded to allow for building without linking to zlib, doesn't seem to be any other reason for this. If we can directly add eBPF into the minimal build, this PR will become a lot smaller.

To be honest i would go for it, so including also `zlib`, in this way we could avoid yet another cmake option :)

Reply:
Just to say this out loud, my initial intention was to split the `MINIMAL_BUILD` into multiple options that could be toggled individually or set them all with `MINIMAL_BUILD`, but desisted because it would've been a lot of work and I only needed the minimal build to run with eBPF. We can discuss this alternative approach if it's something that would be interesting for the community.

Reply:
yeah until we don't have a clear use case I would avoid putting this burden on your shoulders :) If we find alternative use cases i would be happy to consider this splitting you proposed!